components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java

```diff
@@ -243,7 +244,14 @@ private void handleErrorResponse(AuthenticationContext authenticationContext, Re
         }
         value.append('\"');
         response.setHeader(AUTH_HEADER_NAME, value.toString());
-        response.sendError(error);
+        response.setStatus(error);
+        response.setHeader("Content-Type","application/json;charset=UTF-8");
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("message", "Requires authentication");
+        jsonObject.put("documentation_url", "https://docs.wso2.com/display/IS511/apidocs/Authentication-framework-"
```

Comment:
Can we rethink adding the documentation link here?
- In this Authentication valve layer, we can't specifically return doc related to invoked API. 
- If we are pointing to the documentation, then we need to point to the correct API doc version
- What If other solutions have their own documentation pages?

Reply:
This has been resolved with this commit https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/164/commits/d7bca3033297e9d011833946b74a9623db5311f5

Followed the standard error response format mentioned [here](https://datatracker.ietf.org/doc/html/rfc7644#section-3.7.3). This is the same error response format mentioned in the IS docs as well for the SCIM API